controllers/destination_controller.go

```diff
@@ -68,6 +73,11 @@ func (r *DestinationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		logger: logger,
 	}
 
+	if destination.GetCleanup() == v1alpha1.DestinationCleanupAll &&
+		!controllerutil.ContainsFinalizer(destination, destinationCleanupFinalizer) {
+		return addFinalizers(opts, destination, []string{destinationCleanupFinalizer})
+	}
+
```

Comment:
Do we intend for this field to be mutable? If so we might have to support going from `all` -> `none`, which would involve removing the previously added finalizer

Reply:
Yes, you are right. Added the support 👍 Thanks!